gramineproject/gramine
# [python,ci-examples] Support SGX extended product ID

KSS is a new feature that introduces additional fields to identify enclave in build time: ISVEXTPRODID and ISVFAMILYID. These fields are defined the the developer and signed in SIGSTRUCT. Add new fields to the manifest to add support, and support in SGX signing tool.

Add a hello world example with KSS enabled: Product ID fields are configured in the manifest and the enclave prints them. This example must be built with remote attestation.

At this point SGX loader is not updated and there are no sanity checks prior to enclave load. Therefore, if an enclave with KSS is loaded on an unsupported platform, it might crash.

Signed-off-by: Lavi, Nir <nir.lavi@intel.com>



## Description of the changes 

The following fields were added to the manifest:
```
sgx.kss = [true|false] (default: false)
sgx.isvextprodid = "[16-byte hex value]"  (default: "0x00000000000000000000000000000000")
sgx.isvfamily    = "[16-byte hex value]"  (default: "0x00000000000000000000000000000000")
```

if `sgx.kss = true`, SGX signing tool will do the following in SIGSTRUCT:
1. Set `attributes.kss = 1`
2. Set `isvextprodid` and isvfamilyid` with values from the manifest

This is a partial fix for #800 . It does not address the following:
1. CONFIGID handling
2. RA-TLS verification callback changes

## How to test this PR? 

Build and run `kss-helloworld` example on a platform with SGX and KSS support. The following should be in the output:
```
isvextprodid = feca0df0feca0df00df0feca0df0feca
isvfamilyid = ffeeddccbbaa99887766554433221100
```

If the platform has SGX enabled but does not support KSS, enclave should not load
